pkg/config/validation/virtualservice.go

```diff
@@ -142,17 +142,31 @@ func validateHTTPRouteMatchRequest(http *networking.HTTPRoute, routeType HTTPRou
 					if header == nil {
 						errs = appendErrors(errs, fmt.Errorf("header match %v cannot be null", name))
 					}
-
 					if _, ok := header.GetMatchType().(*networking.StringMatch_Prefix); ok {
 						if header.GetPrefix() == "" {
 							errs = appendErrors(errs, fmt.Errorf("header prefix match %v may not be empty", name))
 						}
 					}
-
 					errs = appendErrors(errs, ValidateHTTPHeaderName(name))
 					errs = appendErrors(errs, validateStringMatchRegexp(header, "headers"))
 				}
 
+				for name, header := range match.WithoutHeaders {
+					if header == nil {
+						errs = appendErrors(errs, fmt.Errorf("withoutHeaders match %v cannot be null", name))
+					}
+					if _, ok := header.GetMatchType().(*networking.StringMatch_Prefix); ok {
+						if header.GetPrefix() == "" {
+							errs = appendErrors(errs, fmt.Errorf("withoutHeaders prefix match %v may not be empty", name))
+						}
+					}
+					errs = appendErrors(errs, ValidateHTTPHeaderName(name))
+					// see canBeConvertedToPresentMatch func for details
+					if header.GetRegex() != "*" {
```

Comment:
can we remove this hack and also check the ".*" in the route generator

Reply:
We use the `validateStringMatchRegexp` function to validate, which rejects `*` because it is not a correct regex.

Reply:
Here, when it is `*`, it is not validated and it is considered  as correct.

Reply:
My thought is to replace the "*" with ".*". So with `withoutHeaders: xxx: regex: .*`,  if a request without xxx header present, it is matched.

Not sure if there is any caveats

Reply:
Is n't `*` supported already and we can not change its interpretation now?

Reply:
Some users may have used "*", which may break compatibility if we don't do special processing.